GKE-runner/main.tf

```diff
@@ -0,0 +1,187 @@
+
+#---------------------------
+# get cluster information
+#---------------------------
+
+data "google_container_cluster" "this_cluster" {
+  name     = var.cluster_name
+  location = var.cluster_location
+}
+
+#--------------------------------------------------------------------
+# create service account for cluster nodes and assign them IAM roles
+#--------------------------------------------------------------------
+
+resource "google_service_account" "runner_nodes" {
+  account_id   = "${var.prefix}-nodes-${random_id.node_pool_suffix.hex}"
+  display_name = "GitLab CI Runner"
+}
+
+resource "google_project_iam_member" "this" {
+  for_each = toset(local.runner_node_roles)
+  project  = var.project
+  role     = each.value
+  member   = "serviceAccount:${google_service_account.runner_nodes.email}"
+}
+
+#--------------------------------
+# create runner node pool
+#--------------------------------
+
+resource "random_id" "node_pool_suffix" {
```

Comment:
rename this resource to `random_suffix`. `node_pool_suffix` suggests it's just for node pools